src/components/Verse/AdvancedCopy/utils/copyText.ts

```diff
@@ -0,0 +1,71 @@
+import clipboardCopy from 'clipboard-copy';
+
+import getTextToCopy from './getTextCopy';
+import validateRangeSelection from './validateRangeSelection';
+
+/**
+ * - Validate the range selection
+ * - Copy the text to clipboard
+ * - update the state if succeed or error
+ * - set the error message if there's an error
+ */
+
+const copyText = async ({
+  showRangeOfVerses,
+  rangeEndVerse,
+  rangeStartVerse,
+  shouldCopyFootnotes,
+  shouldCopyText,
+  translations,
+  verseKey,
+}) => {
+  // if a range is selected, we need to validate it first
+  if (showRangeOfVerses) {
+    const validationError = validateRangeSelection(rangeStartVerse, rangeEndVerse);
+    // if the validation fails
+    if (validationError) {
+      throw validationError;
+    }
+  }
+
+  const getTextBlobPromise = () =>
+    getTextToCopy({
+      rangeEndVerse,
+      rangeStartVerse,
+      shouldCopyFootnotes,
+      shouldCopyText,
+      showRangeOfVerses,
+      translations,
+      verseKey,
+    }).then((text) => new Blob([text], { type: 'text/plain' }));
+
+  const textBlobPromise = getTextBlobPromise();
+
+  /**
+   * Safari supports `text/plain` : promise
+   * Chrome doesn't support it.
+   *
+   * Safari needs `navigator.clipboard.write` to be called immediately without waiting
+   * Chrome can wait
+   *
+   * So, for safari we call it navigator.clipboard immediately and give it a promise
+   * for chrome we wait for the promise to resolve, the wen call navigator.clipboard.write
+   */
+
+  try {
+    copyOnSafari(textBlobPromise);
+    return textBlobPromise;
+  } catch (e) {
+    const blob = await textBlobPromise;
```

Comment:
I wonder why we need to do this since [`Navigator.clipboard`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard) should be working on Chrome too according to the Browser compatibility table of that API. Have you tried what is recommened [here](https://stackoverflow.com/questions/57278923/chrome-76-copy-content-to-clipboard-using-navigator)  or [`Clipboard.writeText()`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText) instead of having to convert the text into a blob then write it?